lib/rules/array-element-newline.js

```diff
@@ -0,0 +1,244 @@
+         */
+        function normalizeOptions(options) {
+            const value = normalizeOptionValue(options);
+
+            return { ArrayExpression: value, ArrayPattern: value };
+        }
+
+        /**
+        * Reports that there shouldn't be a line break after the first token
+        * @param {ASTNode} node - The node to report in the event of an error.
+        * @param {Token} token - The token to use for the report.
+        * @returns {void}
+        */
+        function reportNoLineBreak(node, token) {
+            const tokenBefore = sourceCode.getTokenBefore(token, { includeComments: true });
+
+            context.report({
+                node,
+                loc: {
+                    start: tokenBefore.loc.end,
+                    end: token.loc.start
+                },
+                message: "There should be no linebreak here.",
+                data: {
+                    tokenValue: token.value
+                },
+                fix(fixer) {
+                    if (astUtils.isCommentToken(tokenBefore)) {
+                        return null;
+                    }
+
+                    if (!astUtils.isTokenOnSameLine(tokenBefore, token)) {
+                        return fixer.replaceTextRange([tokenBefore.range[1], token.range[0]], " ");
+                    }
+
+                    /*
+                     * This will check if the comma is on the same line as the next element
+                     * Following array:
+                     * [
+                     *     1
+                     *     , 2
+                     *     , 3
+                     * ]
+                     *
+                     * will be fixed to:
+                     * [
+                     *     1, 2, 3
+                     * ]
+                     */
+                    const twoTokensBefore = sourceCode.getTokenBefore(tokenBefore, { includeComments: true });
+
+                    if (astUtils.isCommentToken(twoTokensBefore)) {
+                        return null;
+                    }
+
+                    return fixer.replaceTextRange([twoTokensBefore.range[1], tokenBefore.range[0]], "");
+
+                }
+            });
+        }
+
+        /**
+        * Reports that there should be a line break after the first token
+        * @param {ASTNode} node - The node to report in the event of an error.
+        * @param {Token} token - The token to use for the report.
+        * @returns {void}
+        */
+        function reportRequiredLineBreak(node, token) {
+            let tokenBefore = sourceCode.getTokenBefore(token, { includeComments: true });
+
+            // in case array elements are surrounded by parentheses
+            // the element token does usually not include them.
+            // set token and tokenBefore before these parentheses
+            if (tokenBefore.value === "(") {
```

Comment:
Maybe it would be better to get the comma token first, and then use the first token after the comma. For example:

```js
const previousElement = /* ... */;
const nextElement = /* ... */;

const commaToken = sourceCode.getFirstTokenBetween(previousElement, nextElement, astUtils.isCommaToken);

const firstElementToken = sourceCode.getTokenAfter(commaToken);
```

That way, you can get the first token of the element regardless of how many parentheses surround it.